files/internals/functions.apf

```diff
@@ -318,50 +454,82 @@ if [ -z "$TYPE" ]; then
 fi
 
 if [ ! "$TOS_0" == "" ]; then
-        for i in `echo $TOS_0 | tr ',' ' '`; do
-                i=`echo $i | tr '_' ':'`
-                $IPT $IPT_FLAGS -t mangle -A $TYPE -p tcp --dport $i -j TOS --set-tos 0
-                $IPT $IPT_FLAGS -t mangle -A $TYPE -p udp --dport $i -j TOS --set-tos 0
-        done
+  for i in `echo $TOS_0 | tr ',' ' '`; do
+    i=`echo $i | tr '_' ':'`
+    j=`echo $i | tr ':' '-'`
+    if [ $EXECUTOR == "nft" ]; then
+      $NFT add rule ip mangle $TYPE tcp dport $j counter ip dscp set 0x00
+      $NFT add rule ip mangle $TYPE udp dport $j counter ip dscp set 0x00
+    else
+      $IPT $IPT_FLAGS -t mangle -A $TYPE -p tcp --dport $i -j TOS --set-tos 0
+      $IPT $IPT_FLAGS -t mangle -A $TYPE -p udp --dport $i -j TOS --set-tos 0
+    fi
+  done
 fi
 
 if [ ! "$TOS_2" == "" ]; then
-        for i in `echo $TOS_2 | tr ',' ' '`; do
-                i=`echo $i | tr '_' ':'`
-                $IPT $IPT_FLAGS -t mangle -A $TYPE -p tcp --dport $i -j TOS --set-tos 2
-                $IPT $IPT_FLAGS -t mangle -A $TYPE -p udp --dport $i -j TOS --set-tos 2
-        done
+  for i in `echo $TOS_2 | tr ',' ' '`; do
+    i=`echo $i | tr '_' ':'`
+    j=`echo $i | tr ':' '-'`
+	 if [ $EXECUTOR == "nft" ]; then
+       $NFT add rule ip mangle $TYPE tcp dport $j counter ip dscp set 0x00
+       $NFT add rule ip mangle $TYPE udp dport $j counter ip dscp set 0x00
+	 else
+       $IPT $IPT_FLAGS -t mangle -A $TYPE -p tcp --dport $i -j TOS --set-tos 2
+       $IPT $IPT_FLAGS -t mangle -A $TYPE -p udp --dport $i -j TOS --set-tos 2
+	 fi
+  done
```

Comment:
tested okay,

same as above, needed to add table and chain for mangle PREROUTING

Reply:
640fe51